seaworthy/lowlevel.py

```diff
@@ -0,0 +1,59 @@
+"""
+This is a module for horrible low-level things that we really wish we
+didn't have to build and maintain ourselves.
+"""
+
+import select
+import struct
+from datetime import datetime, timedelta
+
+
+class SocketClosed(Exception):
+    """
+    Exception used for flow control. :-(
+    """
+
+
+def stream_logs(container, stdout=1, stderr=1, stream=1, timeout=10.0):
+    deadline = datetime.now() + timedelta(seconds=timeout)
```

Comment:
Curious as to why you're using `datetime.now()` rather than something like `time.monotonic()`?